roofit/batchcompute/CMakeLists.txt

```diff
@@ -0,0 +1,65 @@
+# Library which powers fast batch computations in Roofit.
+ROOT_STANDARD_LIBRARY_PACKAGE(RooBatchCompute
+  HEADERS
+    RooBatchCompute.h
+    BracketAdapter.h
+    RooSpan.h
+    RooVDTHeaders.h
+    RunContext.h
+  SOURCES
+    src/Initialisation.cxx
+    src/RunContext.cxx
+  DICTIONARY_OPTIONS
+    "-writeEmptyRootPCM"
+  #~  LINKDEF
```

Comment:
To be honest I don't know what I should do here. What is that linkdef file? CMake complains if there isn't one but the project runs perfectly if i just add an empty linkdef.

Reply:
> None of those should have I/O capabilities, so no Linkdef required. You have to ask other people, though, how to run without. Maybe standard library package is the wrong approach, and cmake library does the right thing.

No you can't just add an empty LinkDef file.

 It is used for generation ROOT dictionaries, you can check some information here: https://root.cern/d/interacting-shared-libraries-rootcling.html and https://root.cern/d/selecting-dictionary-entries-linkdefh.html

@manolismih why ROOT_LINKER_LIBRARY didn't work here for you?

Reply:
Why not an empty LinkDef ?
This is the simplest solution to avoid having the dictionary generated for the classes without  avoiding using `ROOT_STANDARD_LIBRARY_PACKAGE`. Also in the future, if one wants to add a new class in the library and have the dictionary it would just require to add a new entry in the LinkDef file

Reply:
I don't understand if are you planning to extend functionality, then you could simply add a Linkdef file afterwards? Please don't forget that `ROOT_STANDARD_LIBRARY_PACKAGE` also generated pcm files. (cc: @Axel-Naumann )

Reply:
Maybe Axel could suggest what to use here `ROOT_STANDARD_LIBRARY_PACKAGE` or `ROOT_LINKER_LIBRARY`?

Reply:
How has this discussion been resolved?

Reply:
My bad, it hasn't... I missed these comments and merged it. I am looking at it now

Reply:
@oshadura My library has some headers that roofit will need. Can a ROOT_LINKER_LIBRARY manage with that?

Reply:
@manolismih I was thinking do you actually need any features that are offered by ROOT_STANDARD_LIBRARY_PACKAGE such as the generation of C++ module (.pcm file) or rootmap files which are used by interpreter to parse headers of libraries that are actually used?

Reply:
@oshadura No i don't need these features, it's an internal use library. But in the example there's no mention to any headers, are these exported to the libraries that depend on it? Is it that ROOT_INSTALL_HEADERS() doing that job?

Reply:
These libraries
- are non-`TObject`s
- do not need a dictionary
- need no PCM, as the headers should never be `#include`d into the interpreter
- as no I/O is required for them,
- shouldn't even have symbols visible outside RooFit
- do not link against other ROOT libraries.
Given all that I think we don't need `ROOT_LINKER_LIBRARY` but just the regular CMake version.

And I bet we don't even need to install the headers.

Reply:
FYI I opened https://github.com/root-project/root/pull/7026 that should resolve this discussion